### ISSUE TITLE
DatePicker autosets value to minDate if value is null, autoOk is true and minDate is greater than current date

### ISSUE THREAD

**Issue by mcMickJuice** [dmtrKovalenko/material-ui-pickers#445]
## Environment

| Tech                | Version |
|---------------------|---------|
| material-ui-pickers |   1.0.0-rc.9      |
| material-ui         |    1.2.0     |
| React               |    16.3.2     |
| Browser             |   Chrome - 66.0.3359.181      |
| Platform            |   OSX      |

## Steps to reproduce

1. set value to null. empty string or undefined, autoOk to true on DatePicker component
2. set minDate to current date + any number of days (i.e. with moment `moment().add('day', 1).format()`
3. Notice that input is empty on initial render
4. Clicking the Calendar to open the modal automatically sets the input value to the value of the minDate.  Modal doesn't open up. Subsequent clicks opens modal like normal

## Expected behavior
Modal should open up like normal and value shouldn't be set on initial opening

## Actual behavior
On initial click to open modal, the value of minDate is set in the input

## Live example
[Example codesandbox](https://codesandbox.io/s/ymj2y48yw1)

**Comment by mcMickJuice** [comment]
**TL;DR; pass in `false` as second parameter to `this.onDateSelect` and `this.props.onChange` in the `Calendar` component**

This is related to `componentDidMount` in Calendar component:

```
  componentDidMount() {
    const {
      date, minDate, maxDate, utils, disableFuture, disablePast,
    } = this.props;

    if (this.shouldDisableDate(date)) { // this is true and is causing onDateSelect to be called
      this.onDateSelect(findClosestEnabledDate({
        date,
        utils,
        minDate,
        maxDate,
        disablePast,
        disableFuture,
        shouldDisableDate: this.shouldDisableDate,
      }));
    }
  }
```

shouldDisableDate is true because the `validateMinMaxDate` condition is returning true:

```
  shouldDisableDate = (day) => {
    const {
      disablePast, disableFuture, shouldDisableDate, utils,
    } = this.props;

    return (
      (disableFuture && utils.isAfterDay(day, utils.date())) ||
      (disablePast && utils.isBeforeDay(day, utils.date())) ||
      this.validateMinMaxDate(day) || // right here is returning true
      shouldDisableDate(day)
    );
  };
```

When this value is true, `this.onDateSelect` is called which bubbles up to `handleChange` which closes the modal and calls the user's onChange:

```
  handleChange = (date, isFinish = true) => {
    this.setState({ date }, () => {
      if (isFinish && this.props.autoOk) {
        this.handleAccept();
        this.wrapper.close();
      }
    });
  }
```

Is the fix just to call `this.onDateSelect` in the Calendar component with `false` as the second parameter?

**Comment by dmtrKovalenko** [comment]
I see. We need to check additionally availability of founded closest day.

**Comment by jstheoriginal** [comment]
This also appears to happen with these props set:

```
<DatePicker
  openToYearSelection
  minDate={moment().add(1, 'days').format(dateFormat)}
  format={dateFormat}
  value={this.state.value || initialValue}
  onChange={this.handleChange}
/>
```

if the user opens up the modal (which starts on the year selection screen), it shows today's date. If they tap ok, it chooses today's date. If they select a year before hitting OK, it will change the date to the min date provided (in this case tomorrow), and they can't choose today.

Today being Thursday, June 7th:

<img width="331" alt="screen shot 2018-06-07 at 2 38 49 pm" src="https://user-images.githubusercontent.com/5117473/41122254-93f5c254-6a60-11e8-8e85-f3a34b278e0f.png">
<img width="326" alt="screen shot 2018-06-07 at 2 38 58 pm" src="https://user-images.githubusercontent.com/5117473/41122263-97c582ac-6a60-11e8-9d93-a8bfc8f83ddc.png">

**Comment by jstheoriginal** [comment]
@mcMickJuice you bet. that's exactly what's happening. I assume that's not how it _should_ work, right?

**Comment by jstheoriginal** [comment]
Based on the PR, once this is released, I'll just need to check for isFinished making sure it's true, correct? I was originally doing that but isFinished was always undefined in current release. good stuff. thanks! 👍
